internal/services/managedredis/managed_redis_resource.go

```diff
@@ -859,3 +880,25 @@ func flattenPersistenceRDB(input *databases.Persistence) string {
 
 	return ""
 }
+
+func isSkuAllowedForScaling(ctx context.Context, clusterClient *redisenterprise.RedisEnterpriseClient, clusterId *redisenterprise.RedisEnterpriseId, targetSkuName string) (warning string, err error) {
+	skusForScaling, err := clusterClient.ListSkusForScaling(ctx, *clusterId)
+	if err != nil {
+		return fmt.Sprintf("SKU scaling cannot be validated due to an error whilst retrieving the list: %+v. The deployment might fail, check resource documentation for more information", err), nil
+	}
+	if skusForScaling.Model.Skus == nil {
+		return "SKU scaling cannot be validated due to Azure returning no information. The deployment might fail", nil
+	}
```

Comment:
Should print a warning and allow the scaling to proceed in here. In this scenario user might not have permission to query valid SKU, or the API has bugs.

```suggestion
	if err != nil {
		log.Printf("[WARN] SKU scaling cannot be validated due to an error whilst retrieving the list. The deployment might fail, check resource documentation for more information: https://learn.microsoft.com/azure/redis/how-to-scale: %+v", err)
		return true
	}
	if skusForScaling.Model.Skus == nil {
		log.Printf("[WARN] SKU scaling cannot be validated due to Azure returning no information. The deployment might fail, check resource documentation for more information: https://learn.microsoft.com/azure/redis/how-to-scale.")
		return true
	}
```

Reply:
Not sure if this kind of warning will be displayed in the same way, but is done.